src/nbl/ext/ImGui/shaders/common.hlsl

```diff
@@ -9,19 +9,48 @@
 	struct PSInput
 	{
 		float4 position : SV_Position;
-		float4 color    : COLOR0;
 		float2 uv : TEXCOORD0;
+		float4 color    : COLOR0;
+		uint drawID : SV_InstanceID;
+		float clip[4] : SV_ClipDistance;
 	};
 
 	struct PushConstants
 	{
+		uint64_t elementBDA;
+		uint64_t elementCount;
 		float2 scale;
 		float2 translate;
+		float4 viewport;
 	};
+	
 #else
 	struct PushConstants
 	{
+		uint64_t elementBDA;
+		uint64_t elementCount;
 		float scale[2];
 		float translate[2];
+		float viewport[4];
 	};
 #endif // __HLSL_VERSION
+
+struct emulated_snorm16_t2
+{
+	#ifdef __HLSL_VERSION
+	float32_t2 unpack() // returns in NDC [-1, 1] range
+	{
+		return clamp(float32_t2(x, y) / 32767.0f, -1.f, +1.f);
+	}
+	#endif
+
+	int16_t x;
+	int16_t y;
+};
```

Comment:
you don't need the `unpack` anymore, the sPIR-V intrinsic you corrected should "just work" (TM)